src/SharedFrameworkValidation/SharedFrameworkValidation.proj

```diff
@@ -56,6 +56,41 @@
              ContinueOnError="ErrorAndStop" />
   </Target>
 
+  <UsingTask TaskName="GetNetCoreAppVersionsFromFile" AssemblyFile="$(BuildToolsTaskDir)Microsoft.DotNet.Build.Tasks.dll"/>
+
+  <Target Name="GetRepoVersionHashes">
+    <GetNetCoreAppVersionsFromFile 
+              PathToVersionsFile="$(PackagesDir)Microsoft.NETCore.App/$(MicrosoftNETCoreAppPackageVersion)/Microsoft.NETCore.App.versions.txt">
+      <Output TaskParameter="VersionHashes" ItemName="VersionHashes" /> 
+    </GetNetCoreAppVersionsFromFile>
+
+    <PropertyGroup>
+      <CoreFxHash Condition="'%(VersionHashes.Identity)' == 'corefx'">%(VersionHashes.VersionHash)</CoreFxHash>
+      <CoreFxCopyLocation>$(BinDir)corefxCopy</CoreFxCopyLocation>
+      <_RepoURL>https://github.com/dotnet/corefx.git</_RepoURL>
+    </PropertyGroup>
+
+    <!-- Windows script -->
+    <ItemGroup>
+      <_CloneRepoLines Include="REM @echo off" />
+      <_CloneRepoLines Include="pushd $(CoreFxCopyLocation.Replace('/', '\'))" />
+      <_CloneRepoLines Include="git clone $(_RepoURL)" />
```

Comment:
I also had this question but it all seems to work just fine. I didn't do a scoped restore because @ericstj brought up a great point which is that tests might be dependent on the tools version, so the way to make sure this won't affect is by cloning again. The only possible complication this brings, is that we might hit maxPath easier, but at least in my machine this works out just fine